.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,62 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master,  releasebranch_7_8]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '36 18 * * 6'
+
+jobs:
+  analyze:
+    name: ${{ matrix.c }} & ${{ matrix.cpp }} - Analyze
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'python' ]
+        c:
+        - gnu99
+        - gnu11
+        - gnu17
+        cpp:
+        - c++11
+        - c++14
+        - c++17
```

Comment:
I am not sure and wanted to test all possibilities. Unfortunately, the Code Scanning alert doesn't show the matrix that was used for creation, so it is not possible to see which combinations led to which alert. But no duplicates are shown, so I guess the job recognizes when an alert already exists.
I would like to reduce to one but am no expert in this. Which one would you recommend? gnu17 + c++17 ? Else I will change to #1288 approach.

Reply:
There is a discussion at [dev-ml](https://lists.osgeo.org/pipermail/grass-dev/2021-January/094899.html) regarding min. standard for G8. No decision has been made yet, but I personally would say now gnu11/c++11 would be sufficient for the time being. This can always be updated. Also no need to overstrain Github's resources with 10s of extra builds.

Reply:
I agree with gnu11/c++11. How relevant the versions are to the CodeQL test? Your comment seems to imply that they influence something, @mmacata. Is there something in the doc about that? If they are not relevant, you could remove them from the matrix and just hardcode them to the `env: CFLAGS...` part. Even if you keep them in the matrix, there is only job based on these (gnu11 & c++11), so no need to include them in the job name (see how the items in Checks look like).

Reply:
I don't know if they are relevant at all. Unfortunately I cannot find any documentation about it. I added them because I thought there might be an indication in the security alerts but as there is none, I will reduce to suggested gnu11/c++11.